docs/CHANGELOG/CHANGELOG-1.12.md

```diff
@@ -30,9 +35,227 @@
     - [Dependencies](#dependencies-1)
     - [Helm Charts](#helm-charts-1)
     - [Instrumentation](#instrumentation-1)
+- [v1.12.0-alpha.1](#v1120-alpha1)
+  - [Downloads for v1.12.0-alpha.1](#downloads-for-v1120-alpha1)
+  - [Changelog since v1.11.0](#changelog-since-v1110)
+  - [Urgent Update Notes](#urgent-update-notes-1)
+  - [Changes by Kind](#changes-by-kind-1)
+    - [API Changes](#api-changes-2)
+    - [Features & Enhancements](#features--enhancements-2)
+    - [Deprecation](#deprecation-2)
+    - [Bug Fixes](#bug-fixes-2)
+    - [Security](#security-2)
+  - [Other](#other-2)
+    - [Dependencies](#dependencies-2)
+    - [Helm Charts](#helm-charts-2)
+    - [Instrumentation](#instrumentation-2)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 
+# v1.12.0
+## Downloads for v1.12.0
+
+Download v1.12.0 in the [v1.12.0 release page](https://github.com/karmada-io/karmada/releases/tag/v1.12.0).
+
+## What's New
+
+### Support Roll Back Migration Safely
+
+This release introduces `PreserveResourcesOnDeletion` field to both PropagationPolicy and ClusterPropagationPolicy API, which provides the ability to rollback migration safely.
+
+The `PreserveResourcesOnDeletion` field controls whether resources should be preserved on the member clusters when the resource template is deleted in the Karmada control-plane. If it is set to true, resources will be preserved on the member clusters.
+
+This feature is particularly useful during workload migration scenarios to ensure that rollback can occur quickly without affecting the workloads running on the member clusters.
+
+For a detailed description of this feature, please refer to [How to Roll Back Migration Operations](https://karmada.io/docs/next/administrator/migration/migrate-in-batch/#how-to-roll-back-migration-operations) and [docs(proposal): Migration Rollback Protection](https://github.com/karmada-io/karmada/pull/5101).
+
+(Feature contributors: @CharlesQQ @XiShanYongYe-Chang @RainbowMango @a7i @chaosi-zju @wulemao)
+
+### Stateful Application Failover Support
+
+This release introduces a new feature for stateful application failover, it provides a generalized way for users to define application state preservation in the context of cluster-to-cluster failovers.
+
+In the previous releases, Karmada’s scheduling logic runs on the assumption that resources that are scheduled and rescheduled are stateless. In some cases, users may desire to conserve a certain state so that applications can resume from where they left off in the previous cluster.
+
+For CRDs dealing with data-processing (such as Flink or Spark), it can be particularly useful to restart applications from a previous checkpoint. That way applications can seamlessly resume processing data while avoiding double processing.
+
+For a detailed description of this feature, please refer to [[Feature] Stateful Application Failover Support](https://github.com/karmada-io/karmada/issues/5788).
```

Comment:
The referencing issue does not provide much useful context for this feature.
@XiShanYongYe-Chang Are you writing the docs of this feature?

Reply:
Writing in progress

Reply:
It's ready now: https://github.com/karmada-io/website/pull/749

Reply:
https://karmada.io/docs/next/userguide/failover/application-failover/#stateful-application-failover-support